commands/inventory/add_agent_qan_postgres_pgstatmonitor_agent.go

```diff
@@ -0,0 +1,108 @@
+	Password              string
+	CustomLabels          string
+	SkipConnectionCheck   bool
+	TLS                   bool
+	TLSSkipVerify         bool
+	QueryExamplesDisabled bool
+}
+
+func (cmd *addAgentQANPostgreSQLPgStatMonitorAgentCommand) Run() (commands.Result, error) {
+	customLabels, err := commands.ParseCustomLabels(cmd.CustomLabels)
+	if err != nil {
+		return nil, err
+	}
+	params := &agents.AddQANPostgreSQLPgStatMonitorAgentParams{
+		Body: agents.AddQANPostgreSQLPgStatMonitorAgentBody{
+			PMMAgentID:           cmd.PMMAgentID,
+			ServiceID:            cmd.ServiceID,
+			Username:             cmd.Username,
+			Password:             cmd.Password,
+			CustomLabels:         customLabels,
+			SkipConnectionCheck:  cmd.SkipConnectionCheck,
+			TLS:                  cmd.TLS,
+			TLSSkipVerify:        cmd.TLSSkipVerify,
+			DisableQueryExamples: cmd.QueryExamplesDisabled,
+		},
+		Context: commands.Ctx,
+	}
+
+	resp, err := client.Default.Agents.AddQANPostgreSQLPgStatMonitorAgent(params)
+	if err != nil {
+		return nil, err
+	}
+	return &addAgentQANPostgreSQLPgStatMonitorAgentResult{
+		Agent: resp.Payload.QANPostgresqlPgstatmonitorAgent,
+	}, nil
+}
+
+// register command
+var (
+	AddAgentQANPostgreSQLPgStatMonitorAgent  = new(addAgentQANPostgreSQLPgStatMonitorAgentCommand)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC = addAgentC.Command("qan-postgresql-pgstatmonitor-agent", "Add QAN PostgreSQL Stat Monitor Agent to inventory").Hide(hide)
+)
+
+func init() {
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Arg("pmm-agent-id", "The pmm-agent identifier which runs this instance").Required().StringVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.PMMAgentID)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Arg("service-id", "Service identifier").Required().StringVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.ServiceID)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Arg("username", "PostgreSQL username for QAN agent").Default("postgres").StringVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.Username)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("password", "PostgreSQL password for QAN agent").StringVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.Password)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("custom-labels", "Custom user-assigned labels").StringVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.CustomLabels)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("skip-connection-check", "Skip connection check").BoolVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.SkipConnectionCheck)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("tls", "Use TLS to connect to the database").BoolVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.TLS)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("tls-skip-verify", "Skip TLS certificates validation").BoolVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.TLSSkipVerify)
+	AddAgentQANPostgreSQLPgStatMonitorAgentC.Flag("disable-queryexamples", "Disable collection of query examples").BoolVar(&AddAgentQANPostgreSQLPgStatMonitorAgent.QueryExamplesDisabled)
```

Comment:
is `disable-queryexamples` or `disable-query-examples`?

Reply:
@percona-csalguero disable-queryexamples because it is same like in MySQLPerfSchema. Or you want change it to disable-query-examples?

Reply:
No. Thanks for the clarification.